90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -977,10 +977,19 @@ populate_be_list() {
       # All other datasets are ignored
       continue
     fi
+    if [ -n "${BOOTFS}" ] && [ "${BOOTFS}" = "${fs}" ] ; then
+      # If BOOTFS is defined, we'll manually append it to the array
+      continue
+    fi
+
     candidates+=( "${fs}" )
   done <<< "$(zfs list -H -o name,mountpoint,org.zfsbootmenu:active)"
 
-  for fs in "${candidates[@]}"; do
+  # Reverse sort the list, to line up with our global usage of --tac
+  readarray -t sorted < <( for env in "${candidates[@]}" ; do echo "$env" ; done | sort -r )
```

Comment:
I believe
```
printf "%s\n" "${candidates[@]}" | sort -r
```
works in place of the shell loop, even if individual entries of `candidates` have spaces.

Reply:
I reverse sorted the list coming out of `zfs list ...`, works like a charm without a pointless duplicated array.